tests/unit/api/metadataSearch.js

```diff
@@ -0,0 +1,258 @@
+const assert = require('assert');
+const querystring = require('querystring');
+
+const async = require('async');
+const { parseString } = require('xml2js');
+
+const metadataSearch = require('../../../lib/api/metadataSearch');
+const { bucketPut } = require('../../../lib/api/bucketPut');
+const objectPut = require('../../../lib/api/objectPut');
+const { cleanup, DummyRequestLogger, makeAuthInfo } = require('../helpers');
+const DummyRequest = require('../DummyRequest');
+
+const { errors } = require('arsenal');
+
+const authInfo = makeAuthInfo('accessKey1');
+const bucketName = 'bucketname';
+const delimiter = '/';
+const log = new DummyRequestLogger();
+const namespace = 'default';
+const postBody = Buffer.from('I am a body', 'utf8');
+const prefix = 'sub';
+
+const objectName1 = `${prefix}${delimiter}objectName1`;
+const objectName2 = `${prefix}${delimiter}objectName2`;
+const objectName3 = 'invalidURI~~~b';
+const objectName4 = `${objectName1}&><"\'`;
+const testPutBucketRequest = new DummyRequest({
+    bucketName,
+    headers: {},
+    url: `/${bucketName}`,
+    namespace,
+}, Buffer.alloc(0));
+const testPutObjectRequest1 = new DummyRequest({
+    bucketName,
+    headers: { 'x-amz-meta-test': 'test' },
+    url: `/${bucketName}/${objectName1}`,
+    namespace,
+    objectKey: objectName1,
+}, postBody);
+const testPutObjectRequest2 = new DummyRequest({
+    bucketName,
+    headers: { 'x-amz-meta-test': 'test' },
+    url: `/${bucketName}/${objectName2}`,
+    namespace,
+    objectKey: objectName2,
+}, postBody);
+const testPutObjectRequest3 = new DummyRequest({
+    bucketName,
+    headers: {},
+    url: `/${bucketName}/${objectName3}`,
+    namespace,
+    objectKey: objectName3,
+}, postBody);
+const testPutObjectRequest4 = new DummyRequest({
+    bucketName,
+    headers: {},
+    url: `/${bucketName}/${objectName3}`,
+    namespace,
+    objectKey: objectName4,
+}, postBody);
+
+const baseGetRequest = {
+    bucketName,
+    namespace,
+    headers: { host: '/' },
+};
+const baseUrl = `/${bucketName}`;
+
+const tests = [
+    {
+        name: 'list of all objects if no delimiter specified',
+        request: Object.assign({ query: {}, url: baseUrl }, baseGetRequest),
+        assertion: result => {
+            assert.strictEqual(result.ListBucketResult.Contents[1].Key[0],
+                objectName1);
+            assert.strictEqual(result.ListBucketResult.Contents[2].Key[0],
+                objectName2);
+            assert.strictEqual(result.ListBucketResult.Contents[0].Key[0],
+                objectName3);
+        },
+    },
```

Comment:
As far as I understand this, we are re-defining tests we already have in `tests/unit/api/bucketGet.js`. While some are using search, most of them are not. Can we clean the `tests/unit/api/bucketGet.js` to remove the search tests and this file to remove the non-search tests?

See: https://github.com/scality/cloudserver/blob/8a91dfef09659d6282a64e47ccc0a2200b3c3d82/tests/unit/api/bucketGet.js#L71

Reply:
thanks to your comment, I found in-memory db doesn't support metadata search. since we already have[ functional tests ](https://github.com/scality/cloudserver/tree/development/8.3/tests/functional/aws-node-sdk/test/mdSearch)and [search query unit tests](https://github.com/scality/cloudserver/blob/development/8.3/tests/unit/utils/validateSearch.js). I just removed the whole unit test file for metadata search api, there is no use to unit test things other than metadata search for this api.

Reply:
Nice finding, and agree :slightly_smiling_face: